### ISSUE TITLE
BT Pay

### ISSUE THREAD

**Issue by AragornDR** [PrivSec-dev/banking-apps-compat-report#83]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

BT Pay

### Link to app

https://play.google.com/store/apps/details?id=ro.btrl.pay

### App version

1.18.1(cfb0d3402

### Country of the app

Romania

### Build Number

SP1A.210812.016.C1.2022051100

### Device

- [ ] Pixel 6 Pro
- [ ] Pixel 6
- [ ] Pixel 5a
- [ ] Pixel 5
- [ ] Pixel 4a (5G)
- [ ] Pixel 4a
- [ ] Pixel 4 XL
- [ ] Pixel 4
- [ ] Pixel 3a XL
- [ ] Pixel 3a
- [ ] Pixel 3 XL
- [X] Pixel 3

### Google Play installed

- [X] Installed
- [ ] Not installed

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### SafetyNet Enforcement

- [X] Enforced
- [ ] Not enforced
- [ ] Unsure

### Native code debugging

- [X] Enabled
- [ ] Disabled

### Stock OS compatibility

- [ ] Works
- [ ] Does not work
- [X] Not tested

### Profile app tested in

- [ ] Main user profile
- [X] Secondary user profile
- [ ] I used a work profile device manager app (Please mention more details about it down below like the app)

### Description of the app's functionality

Both transferring money and paying with nfc work flawlessly. I don't know if AliasPay works, as I don't use it.

**Comment by akc3n** [comment]
@AragornDR hey, thank you very much! I'll get that added this evening.

**Comment by GotRobbd** [comment]
As of now, with the latest update to 3.6.1, the app no longer opens and throws a "security error" which leaves me with no other choice, but to uninstall the app. After reinstalling it from Play Store, it does the same. Enabling exploit protection compatibility mode does nothing.

**Comment by GotRobbd** [comment]
[https://discuss.grapheneos.org/d/18208-btpay-ro-mobile-banking-app-not-working/11](https://discuss.grapheneos.org/d/18208-btpay-ro-mobile-banking-app-not-working/11)

I am not sure if they might have changed anything, but I managed to set up the app with no issues, and it has been working fine since then. The caveat to it is it needs native code debugging on to work properly. Could anyone have a look over it again?

**Comment by spring-onion** [comment]
@GotRobbd Are you that user on the forum? It still doesn't work on my end.

**Comment by GotRobbd** [comment]
Yes that's me.

**Comment by spring-onion** [comment]
@GotRobbd To me that sounds like the support staff couldn't figure out what went wrong and assumed some bug had raised eyebrows. I was greeted by that same security error when I retried the app a few days ago so whatever checks they have in place were not modified. It's some selective thing that they can adjust on an individual level, so while it's great you managed to persuade them, and other people may also have success with that in the future, it's unclear if they would be understanding of you using GrapheneOS. Also you mentioned the recent reviews complaining about this problem were from pixel users in general so this might not be related to GrapheneOS at all.

**Comment by AragornDR** [comment]
I also encountered this message while the application was installed on a secundary profile and in the private space. I managed to make it work after installing google services on the main profile, without having to call the bank.

**Comment by spring-onion** [comment]
Thanks for chiming in, will mark compatible again!

**Comment by evo11x** [comment]
Version 4.1.1 from August, no longer works on Pixel 8 Pro

**Comment by spring-onion** [comment]
What if you...

1. Disable the exploit protection compatibility mode.
2. Disable secure app spawning (found only in the owner profile).
3. Reboot your phone.

**Comment by spring-onion** [comment]
@evo11x In case you haven't seen, apologies if you have.

**Comment by evo11x** [comment]
@spring-onion I have disable the exploit protection and secure app spawning, rebooted the phone and it still does not work, it says device is rooted.

**Comment by GotRobbd** [comment]
Their app is getting ridiculous. I recently had the same issue being locked out due to "security reasons", called their customer service, got the problem issued, updated the app, and now I'm back to square one. Interestingly, if you create a new profile, install Google Play Services alongside the app and launch it the app works just fine.

**Comment by Gwirarien** [comment]
Calling support doesn’t really help much unless they submit an expedite request, which usually only happens after you call them a second time. After that, they typically call back the next business day.

I’ve contacted customer support 3-4 times in the last two months, and each time they unblocked the app based on the device ID. The fix only lasted about a week the last time. Support seemed doubtful and a bit frustrated since this isn’t the first time I’ve reported it. They asked whether I had any remote access or surveillance apps (like TeamViewer) installed, and if developer options were enabled or the phone was rooted, which I don't. Their last suggestion was to "take the phone to a service center for a checkup". I didn't mention that I’m running GrapheneOS.

Now I’m locked out again. I’m not sure what triggers it, but it's random and the only workaround I’ve found so far (apart from calling support) is to create a new user profile each time the issue happens, then the app works fine for a while before breaking again. At this point, it feels like the only alternatives are to keep doing that or to switch banks entirely.

**Comment by straightcurve** [comment]
hey, i noticed that if i set the default wallet app to "None", it works. it asks you to set it every time you open up the app, but just ignore it and it'll be fine. i have been using it for a month now with no issues.

**Comment by terrorbyte69420** [comment]
> hey, i noticed that if i set the default wallet app to "None", it works. it asks you to set it every time you open up the app, but just ignore it and it'll be fine. i have been using it for a month now with no issues.

Just tried that, updated it too since it was out of date apparently, no dice.

**Comment by popeliv** [comment]
The app was working just fine for my until recently, until I also got a security notice prompting me to reinstall the app. The app reinstalled and runs fine, but I can't login to my account because it refuses to read the sms code from my messages (manual input is disabled). I've checked and double checked permissions, I tried giving it every permission possible but nothing seems to work.

**Comment by spring-onion** [comment]
@popeliv It probably relies on Google Play to give it the code, give that permission to access SMS and it should work.

**Comment by popeliv** [comment]
@spring-onion I tried that. I tried giving google play and google play services all permissions, I tried running in exploit compatibility mode, restarting, etc. It's been like this for about a week so I had time to try all sorts of stuff.

Also, It worked before, when I first installed the app in August. Then, after I got the security dialog and reinstalled the app it wouldn't work anymore.

**Comment by GotRobbd** [comment]
I think you could lower the risks of the app complaining about anything it doesn't like. I installed it in a separate profile, with GMS too, exploit protections off, and if the app wouldn't let you fill in the SMS code, giving SMS permissions to Play Services. I think keeping native code debugging off especially, and having a clean profile prevents it from throwing the error mentioned above.

**Comment by marcg03** [comment]
by creating a separate profile it worked for my pixel 9a for both BT Pay and BT Go. I did a reboot after installing google play store and logging in and only afterwards installed the banking apps.

**Comment by almara17** [comment]
I am on Pixel 8a, installed Google Play Store and Play Services (and gave it SMS and Network) in a separated profile and then BT Pay.  

BT Pay version  4.5.1 with Exploit protection compatibility mode set to on.

It works without issues.

**Comment by iloveakitas** [comment]
Just installed the app and it works, had no issues with it except for nfc pay, it doesn't work and I'm not sure why, can anyone say if nfc can work in any way in bt pay?

**Comment by marcg03** [comment]
Yeah I do use nfc and didn't do anything special besides enable it directly in the settings.

**Comment by iloveakitas** [comment]
Fixed nfc payment issue just by removing cache/storage data and logging in again so for me the app fully works now
